src/lib/core/Global.h

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/core/CHIPConfig.h>
+
+#include <new>
+
+namespace chip {
+
+/**
+ * A wrapper for global object that enables initialization and destruction to
+ * be configured by the platform via `CHIP_CONFIG_GLOBALS_*` options.
+ *
+ * The contained object of type T is default constructed, possibly lazily.
+ *
+ * This class is generally NOT thread-safe; external synchronization is required.
+ */
+template <class T>
+class Global
+{
+public:
+    /// Returns the global object, initializing it if necessary.
+    T & operator*() { return get(); }
```

Comment:
Worth also having a `operator T & () { return get(); }` and the const version of that?  Would allow getting rid of a bunch of `*` this PR adds.

Reply:
I think that's too "magical"... I'd like to see something at the call site that makes it obvious or at least hints that there is more than just a variable access going on.